### PR TITLE
Add docker step to chef cloud build

### PR DESCRIPTION
**Pull request by FernandoLGuzman**
#### Change overview
Added an additional step at the end of the chef cloud build. This step will call a script to dockerize chef binaries and then push them to a GCR bucket.

### CODE REVIEW

File: examples/chef/create_docker.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+
+set -e # Exit on any script error.
+
+IMAGE_NAME='us.gcr.io/nest-eng/chef/dimmable-light'
```
Comment (andy31415) [inline examples/chef/create_docker.sh:5]: This seems a bit hardcoded:

- why is the image nest-eng? Can we add it as an argument (so at least cloudbuild sets it but we make the script generic enough to be an OSS component)?
- file requires a license blurb (see other shell scripts)
- same comment regarding hardcoding for /workspace and the name of linux_..... .

**Review by andy31415 — DISMISSED** [review]
Need to remove hardcoding and make this a bit cleaner.
Consider placing scripts into integrations/cloudbuild rather than examples/chef since this seems very tightly coupled with cloudbuild.

File: examples/chef/create_docker.sh (hunk 1 above)
Comment (FernandoLGuzman) [reply examples/chef/create_docker.sh]: Removed the hardcoding from this script and made use of Substitution Variables from the Google Cloud build to make this more generic.

**Review by andy31415 — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

**Comment by bzbarsky-apple** [comment]
@FernandoLGuzman Restyle CI is failing.

**Pull request merged**